src/components/Capture/capture.js

```diff
@@ -85,29 +85,40 @@ class Capture extends Component {
     this.validateCapture(message.id, valid)
   }
 
-  handleBase64 = (lossyBase64, base64) => {
-    const blob = base64toBlob(base64)
-    this.handleImage(lossyBase64, blob, base64)
-  }
-
-  handleImage = (base64ImageLossy, image, base64) => {
-    if (!image) {
+  handleImage = (lossyBase64, blob) => {
+    if (!blob) {
       console.warn('Cannot handle a null image')
       return;
     }
-    const payload = this.createPayload(image, base64ImageLossy, base64)
-    functionalSwitch(this.props.method, {
-      document: ()=> this.handleDocument(payload),
-      face: ()=> this.handleFace(payload)
-    })
+
+    const handleImageInternal = (lossyBase64, blob, base64) => {
```

Comment:
The next 20 lines of logic still feels awkward to me. When would we have a lossyBase64 and a base64 argument? If the answer is never then can't this function just have two arguments (blob, base64)?